File: parse_html.js
const fs = require('fs');
const path = require('path');

const filePath = 'C:\\Users\\ideba\\.gemini\\antigravity\\brain\\50d55630-60b4-4900-821b-b79dafc5cd3a\\.system_generated\\steps\\6\\content.md';

let htmlContent = fs.readFileSync(filePath, 'utf8');

if (htmlContent.startsWith('Title:')) {
    const parts = htmlContent.split('---');
    if (parts.length > 1) {
        htmlContent = parts.slice(1).join('---');
    }
}

// In ChatGPT shared links, conversation data is usually embedded inside the html, often under a script with id="__NEXT_DATA__" or in the state of the app.
// Let's write a simple regex to extract __NEXT_DATA__ content.
const nextDataMatch = htmlContent.match(/<script id="__NEXT_DATA__" type="application\/json">([\s\S]*?)<\/script>/);

if (nextDataMatch) {
    try {
        const data = JSON.parse(nextDataMatch[1]);
        console.log("Found __NEXT_DATA__ JSON!");
        
        // Let's inspect the structure and extract messages
        const props = data.props || {};
        const pageProps = props.pageProps || {};
        const sharedResponse = pageProps.sharedResponse || {};
        const serverResponse = pageProps.serverResponse || {};
        
        let conversation = sharedResponse.conversation || serverResponse.conversation || null;
        if (!conversation && pageProps.sharedResponse && pageProps.sharedResponse.conversation) {
            conversation = pageProps.sharedResponse.conversation;
        }

        // Sometimes the conversation is nested under pageProps.serverResponse.data.conversation
        if (!conversation && pageProps.serverResponse && pageProps.serverResponse.data) {
            conversation = pageProps.serverResponse.data.conversation;
        }
        
        if (conversation) {
            const messages = [];
            const mapping = conversation.mapping || {};
            for (const key in mapping) {
                const node = mapping[key];
                if (node && node.message && node.message.content) {
                    const author = node.message.author ? node.message.author.role : 'unknown';
                    const parts = node.message.content.parts || [];
                    const text = parts.join('\n');
                    if (text && text.trim()) {
                        messages.push({ author, text });
                    }
                }
            }
            fs.writeFileSync('extracted_messages.json', JSON.stringify(messages, null, 2), 'utf8');
            console.log(`Extracted ${messages.length} messages to extracted_messages.json`);
            
            // Also write a human readable version
            let readable = "";
            messages.forEach((msg, index) => {
                readable += `=== Message ${index + 1} (${msg.author.toUpperCase()}) ===\n${msg.text}\n\n`;
            });
            fs.writeFileSync('extracted_messages.txt', readable, 'utf8');
            console.log("Created readable extracted_messages.txt");
            process.exit(0);
        } else {
            console.log("Could not find conversation object in __NEXT_DATA__ props. Keys in pageProps:", Object.keys(pageProps));
        }
    } catch (e) {
        console.error("Failed to parse __NEXT_DATA__:", e);
    }
} else {
    console.log("No __NEXT_DATA__ script tag found.");
}

// Fallback: simple regex to find divs with message classes or text contents
console.log("Attempting fallback text extraction...");
const cleanHtml = htmlContent.replace(/<script[\s\S]*?<\/script>/gi, '').replace(/<style[\s\S]*?<\/style>/gi, '');
const textMatches = [];
// Find paragraph tags or div content that looks like text
const regex = /<(div|p|span|h1|h2|h3)[^>]*>(.*?)<\/\1>/gi;
let match;
while ((match = regex.exec(cleanHtml)) !== null) {
    let txt = match[2].replace(/<[^>]*>/g, '').trim();
    if (txt.length > 20 && !txt.includes('{') && !txt.includes('}')) {
        textMatches.push(txt);
    }
}
fs.writeFileSync('fallback_text.txt', textMatches.join('\n\n'), 'utf8');
console.log("Fallback text matches written to fallback_text.txt");

File: parse_html.py
import json
import re
from bs4 import BeautifulSoup

file_path = r"C:\Users\ideba\.gemini\antigravity\brain\50d55630-60b4-4900-821b-b79dafc5cd3a\.system_generated\steps\6\content.md"

with open(file_path, "r", encoding="utf-8") as f:
    html_content = f.read()

# Strip any leading markdown header text if present
if html_content.startswith("Title:"):
    # Split by the first ---
    parts = html_content.split("---", 1)
    if len(parts) > 1:
        html_content = parts[1]

soup = BeautifulSoup(html_content, "html.parser")

# Look for NEXT_DATA or script elements containing JSON share data
# Usually ChatGPT shares have data in a script tag with id="self" or __NEXT_DATA__
# or we can extract the main conversation container text.
# Let's first look for __NEXT_DATA__
next_data = soup.find("script", id="__NEXT_DATA__")
if next_data:
    try:
        data = json.loads(next_data.string)
        # Find shared conversation
        print("Found __NEXT_DATA__ JSON structure!")
        with open("e:\\AntiGravity\\next_data.json", "w", encoding="utf-8") as out:
            json.dump(data, out, indent=2)
    except Exception as e:
        print("Failed to parse __NEXT_DATA__ JSON:", e)

# Let's extract all main text content or print it nicely
# ChatGPT messages are typically inside divs with class/data tags
# Or we can just get the text and clean it up.
print("Extracting all text from divs...")
messages = []
# Let's look for elements with class containing 'message' or 'markdown'
for div in soup.find_all(class_=re.compile("markdown|message|whitespace-pre-wrap")):
    text = div.get_text(separator="\n").strip()
    if text and text not in messages:
        messages.append(text)

with open("e:\\AntiGravity\\extracted_messages.txt", "w", encoding="utf-8") as out:
    for i, msg in enumerate(messages):
        out.write(f"--- Message {i+1} ---\n{msg}\n\n")

print(f"Extracted {len(messages)} messages to extracted_messages.txt")
